spot-ml/src/test/scala/org/apache/spot/lda/SpotLDAWrapperTest.scala

```diff
@@ -59,47 +61,92 @@ class SpotLDAWrapperTest extends TestingSparkContextFlatSpec with Matchers {
+    logger.setLevel(Level.WARN)
+
+    val ldaAlpha =  1.2
+    val ldaBeta = 1.001
+    val ldaMaxIterations = 20
+
+    val optimizer = "em"
+
+    val catFancy = SpotLDAInput("cat fancy", "cat", 1)
+    val dogWorld = SpotLDAInput("dog world", "dog", 1)
+
+    val data = sparkContext.parallelize(Seq(catFancy, dogWorld))
+    val out = SpotLDAWrapper.runLDA(sparkContext, sqlContext, data, 2, logger, Some(0xdeadbeef), ldaAlpha, ldaBeta,
+      optimizer ,ldaMaxIterations, FloatPointPrecisionUtility64)
+
+    val topicMixDF = out.docToTopicMix
+    val dogTopicMix: Array[Double] =
+      topicMixDF.filter(topicMixDF(DocumentName) === "dog world").select(TopicProbabilityMix).first()
+        .toSeq(0).asInstanceOf[Seq[Double]].toArray
+
+    val catTopicMix: Array[Double] =
+      topicMixDF.filter(topicMixDF(DocumentName) === "cat fancy").select(TopicProbabilityMix).first()
+        .toSeq(0).asInstanceOf[Seq[Double]].toArray
+
+    val catTopics = out.wordResults("cat")
+    val dogTopics = out.wordResults("dog")
+
+    Math.abs(1 - (catTopicMix(0) * catTopics(0) + catTopicMix(1) * catTopics(1))) should be < 0.01
+    Math.abs(1 - (dogTopicMix(0) * dogTopics(0) + dogTopicMix(1) * dogTopics(1))) should be < 0.01
+  }
+
+  it should "handle an extremely unbalanced two word doc with Online optimizer" in {
     val logger = LogManager.getLogger("SuspiciousConnectsAnalysis")
     logger.setLevel(Level.WARN)
 
+    val ldaAlpha =  0.0009
+    val ldaBeta = 0.00001
+    val ldaMaxIterations = 400
+
+    val optimizer = "online"
+
     val catFancy = SpotLDAInput("pets", "cat", 1)
     val dogWorld = SpotLDAInput("pets", "dog", 999)
 
     val data = sparkContext.parallelize(Seq(catFancy, dogWorld))
     val out = SpotLDAWrapper.runLDA(sparkContext, sqlContext, data, 2, logger, Some(0xdeadbeef), ldaAlpha, ldaBeta,
-      ldaMaxIterations, FloatPointPrecisionUtility32)
+      optimizer, ldaMaxIterations, FloatPointPrecisionUtility64)
 
     val topicMixDF = out.docToTopicMix
 
     val topicMix =
       topicMixDF.filter(topicMixDF(DocumentName) === "pets").select(TopicProbabilityMix).first().toSeq(0)
-        .asInstanceOf[Seq[Float]].toArray
+        .asInstanceOf[Seq[Double]].toArray
     val catTopics = out.wordResults("cat")
     val dogTopics = out.wordResults("dog")
 
-    Math.abs(topicMix(0).toDouble * catTopics(0) + topicMix(1).toDouble * catTopics(1)) should be < 0.01
-    Math.abs(0.999 - (topicMix(0).toDouble * dogTopics(0) + topicMix(1).toDouble * dogTopics(1))) should be < 0.01
+    Math.abs(topicMix(0) * catTopics(0) + topicMix(1) * catTopics(1)) should be < 0.01
+    Math.abs(0.999 - (topicMix(0) * dogTopics(0) + topicMix(1) * dogTopics(1))) should be < 0.01
   }
 
-  it should "handle distinct docs on distinct words" in {
+  it should "handle distinct docs on distinct words with Online optimizer" in {
     val logger = LogManager.getLogger("SuspiciousConnectsAnalysis")
     logger.setLevel(Level.WARN)
+
```

Comment:
After all these changes are done I still have pending the alpha and beta tuning. I still think these paramaters are quite low if we consider that the default for both parameters is 1.0/K. 

I also suggest for the future to call the parameters: docConcentration and topicConcentration to avoid confusion. Even the LDA committers recognize that different papers call the parameters differently.